fpdf/text_region.py

```diff
@@ -0,0 +1,428 @@
+            # line height doesn't change much within the text block.
+            # The "correct" solution would require an exact precalculation of the hight of
+            # each column with the specific line heights and iterative regrouping of lines,
+            # which seems excessive at this point.
+            # Contribution of a more reliable but still reasonably simple algorithm welcome.
+            page_bottom = bottom
+            if not text_lines:
+                return
+            tot_height = sum(l.line.height for l in text_lines)
+            col_height = tot_height / self._ncols
+            avail_height = bottom - top
+            if col_height < avail_height:
+                balancing = True  # We actually have room to balance on this page.
+                # total height divided by n
+                bottom = top + col_height
+                # A bit more generous: Try to keep the rightmost column the shortest.
+                lines_per_column = math.ceil(len(text_lines) / self._ncols) + 0.5
+                mult_height = text_lines[0].line.height * lines_per_column
+                if mult_height > col_height:
+                    bottom = top + mult_height
+                if bottom > page_bottom:
+                    # Turns out we don't actually have enough room.
+                    bottom = page_bottom
+                    balancing = False
+        for c in range(self._cur_column, self._ncols):
+            if not text_lines:
+                return
+            if c != self._cur_column:
+                self._cur_column = c
+            col_left, col_right = self.current_x_extents(0, 0)
+            if self.pdf.x < col_left or self.pdf.x >= col_right:
+                self.pdf.x = col_left
+            if balancing and c == (self._ncols - 1):
+                # Give the last column more space in case the balancing is out of whack.
+                bottom = self.pdf.h - self.pdf.b_margin
+            last_line_height = self._render_column_lines(text_lines, top, bottom)
+            if balancing:
+                new_y = self.pdf.y + last_line_height
+                if new_y > next_y:
+                    next_y = new_y
+        if balancing:
+            self.pdf.y = next_y
+
+    def render(self):
+        if not self._paragraphs:
+            return
+        text_lines = self.collect_lines()
+        if not text_lines:
+            return
+        page_bottom = self.pdf.h - self.pdf.b_margin
+        _first_page_top = max(self.pdf.t_margin, self.pdf.y)
+        self._render_page_lines(text_lines, _first_page_top, page_bottom)
+        while text_lines:
+            self.pdf.add_page(same=True)
+            self._cur_column = 0
+            self._render_page_lines(text_lines, self.pdf.y, page_bottom)
+
+    def current_x_extents(self, y, height):
+        left, right = self._cols[self._cur_column]
+        return left, right
+
+
+class LWrapper(NamedTuple):
```

Comment:
```suggestion
class LineWrapper(NamedTuple):
```
Does "L" stands for "Line"? If so, I think it would be clearer to avoid this abbreviation.

Reply:
Looks fine (but just this one change here won't do, so I won't commit it here).